Matrix.py

```diff
@@ -0,0 +1,168 @@
+        """subtracting a matrix from another
+                    args: self.data (tuple)- A matrix
+                        other(tuple) -A matrix
+
+                    returns: return the new matrix (tuple)
+
+                     raises: syntaxerror if "other" isn't a matrix"""
+        if type(other) is Matrix:
+            answer = []
+            for line in range(len(self.data)):
+                answer.append([])
+                for num in range(len(self.data)):
+                    answer[line].append(self.data[line][num] - other[line][num])
+            # converts list of lists to type Matrix. ls are the inside lists
+            return Matrix((tuple([tuple(ls) for ls in answer])))
+        else:
+            raise SyntaxError("only matrix's subscription")
+
+    def __truediv__(self, other):
+        """divides a matrix by scalar
+                    args: self.data (tuple)- A matrix
+                        other(int/float) -A scalar
+
+                    returns: return the new matrix (tuple)
+
+                     raises: syntaxerror if "other" isn't a scalar"""
+        if type(other) is float or int:  # the program doesnt support division by other Matrix
+            return self.scalar_mul(1 / other)
+        else:
+            raise SyntaxError("only scalar division")
+
+    def __mul__(self, other):
+        """mul function decides which multiply method is needed for each call
+        args: self.data(matrix)- the first factor
+         other(int/float/matrix)- the second factor
+         no returns
+         exceptions: SyntaxError if other isn't a matrix or a scalar"""
+        if type(other) is Matrix:
+            return self.matrices_mul(other)
+        elif type(other) is float or type(other) is int:
+            return self.scalar_mul(other)
+        else:
+            raise SyntaxError("invalid input")
+
+    def matrices_mul(self, other):
+        """multiply a matrix by matrix
+                    args: self.data (tuple)- A matrix
+                        other(tuple) -A matrix
+                    returns: return the new matrix (tuple)"""
+        answer = []
+        new_other = []
+        # this loop convert "other" from Matrix type to a list of lists named "new_other".
+        for line in range(len(self.data)):
+            new_other.append([])
+            answer.append([0] * len(self.data))
+            for num in range(len(self.data)):
+                new_other[line].append(other[num][line])
+        # this is the multiplication process->
+        for line in range(len(self.data)):
+            for num in range(len(self.data)):
+                for i in range(len(self.data)):
+                    answer[line][num] += self.data[line][i] * new_other[num][i]
+#                   here the program adds to the zero the summery of the multiplications of
+#                   any two numbers which their places in the matrices are fits by the matrices
+#                   multiplication rules.
```

Comment:
fix hashtags location

Reply:
Igal i think you are checking the wrong version